src/uct/sm/mm/mm_def.h

```diff
@@ -36,4 +38,22 @@ enum {
           (uct_mm_recv_desc_t *) ((_fifo_elem_p)->desc_chunk_base_addr +  \
           (_fifo_elem_p)->desc_offset - (_iface)->rx_headroom) - 1;
 
+typedef struct uct_mm_pd_config {
+    uct_pd_config_t      super;
+    ucs_ternary_value_t  hugetlb_mode;     /* Enable using huge pages */
+} uct_mm_pd_config_t;
+
+typedef struct uct_sysv_pd_config {
+    uct_mm_pd_config_t      super;
+} uct_sysv_pd_config_t;
+
+typedef struct uct_posix_pd_config {
+    uct_mm_pd_config_t      super;
+    char                    *path;
+} uct_posix_pd_config_t;
+
+typedef struct uct_xpmem_pd_config {
+    uct_mm_pd_config_t      super;
+} uct_xpmem_pd_config_t;
+
```

Comment:
yes but these structs are needed in mm_pd.h which includes mm_def.h.
i thought that setting the typedefs in an .h file would be more correct. 
i can create an .h file per each mm type but it would only have these structs.
do you think it should be moved?